fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/control/FhirSchemaGenerator.java

```diff
@@ -367,31 +369,72 @@ public void buildSchema(PhysicalDataModel model) {
                 () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_CODE_SYSTEM.toLowerCase() + ".sql", null),
                 Arrays.asList(fhirSequence, codeSystemsTable, allTablesComplete),
-                procedurePrivileges);
+                procedurePrivileges,
+                DbType.DB2);
         pd.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
 
         pd = model.addProcedure(this.schemaName,
                 ADD_PARAMETER_NAME,
                 FhirSchemaConstants.INITIAL_VERSION,
                 () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_PARAMETER_NAME.toLowerCase() + ".sql", null),
                 Arrays.asList(fhirSequence, parameterNamesTable, allTablesComplete),
-                procedurePrivileges);
+                procedurePrivileges,
+                DbType.DB2);
         pd.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
 
         pd = model.addProcedure(this.schemaName,
                 ADD_RESOURCE_TYPE,
                 FhirSchemaConstants.INITIAL_VERSION,
                 () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_RESOURCE_TYPE.toLowerCase() + ".sql", null),
                 Arrays.asList(fhirSequence, resourceTypesTable, allTablesComplete),
-                procedurePrivileges);
+                procedurePrivileges,
+                DbType.DB2);
         pd.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
 
         pd = model.addProcedure(this.schemaName,
                 ADD_ANY_RESOURCE,
                 FhirSchemaConstants.INITIAL_VERSION,
                 () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_ANY_RESOURCE.toLowerCase() + ".sql", null),
                 Arrays.asList(fhirSequence, resourceTypesTable, allTablesComplete),
-                procedurePrivileges);
+                procedurePrivileges,
+                DbType.DB2);
+        pd.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
+
+        // Add stored procedures/functions for postgresql.
+        pd = model.addProcedure(this.schemaName,
+                ADD_CODE_SYSTEM,
+                FhirSchemaConstants.INITIAL_VERSION,
+                () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_CODE_SYSTEM.toLowerCase()
+                        + "_" + DbType.POSTGRESQL.value() + ".sql", null),
+                Arrays.asList(fhirSequence, codeSystemsTable, allTablesComplete),
+                procedurePrivileges, DbType.DB2);
+        pd.addTag(SCHEMA_GROUP_TAG, FHIRDATA_GROUP);
+
+        pd = model.addProcedure(this.schemaName,
+                ADD_PARAMETER_NAME,
+                FhirSchemaConstants.INITIAL_VERSION,
+                () -> SchemaGeneratorUtil.readTemplate(adminSchemaName, schemaName, ADD_PARAMETER_NAME.toLowerCase()
+                        + "_" + DbType.POSTGRESQL.value() + ".sql", null),
+                Arrays.asList(fhirSequence, parameterNamesTable, allTablesComplete),
+                procedurePrivileges, DbType.DB2);
```

Comment:
it says postgres above, why does it state db2 on this line.

Reply:
typo error, fixed.